### PR TITLE
feat: Remove filter from `createComponentData()`

### LINKED ISSUE #6431: Cms state deprecation - remove filter from `createComponentData()`
Search for `TODO(issue:6431)` in `projects/core/src/cms/facade/cms.service.ts` for the exact spot.

The reason for removal: with `filter` in place, when moving to a page that has restrictions, the component data will still emit the previous value.
Removing it causes some components to fail, because they are not checking if the data is actually there. I noticed these that this component is failing, but there are possibly more:
 - `tab-paragraph-container.component.ts` when visiting any PDP page

### PR DESCRIPTION
**Pull request by dunqan**
Closes #6431

### CODE REVIEW

File: projects/core/src/cms/store/selectors/components.selectors.ts [hunk 1]
```diff
@@ -36,37 +36,55 @@ export const componentsLoaderStateSelectorFactory = (
...
   return createSelector(componentsContextSelectorFactory(uid), state =>
     state ? state.component : undefined
   );
 };
 
+/**
+ * This selector will return:
+ *   - CmsComponent instance if we have component data for specified context
+ *   - null if there is no component data for specified context
+ *   - undefined status of component data for specified context is unknown
```
Comment (Platonn) [inline projects/core/src/cms/store/selectors/components.selectors.ts]:
```suggestion
 *   - undefined if the exists status of component data for specified context is unknown
```

File: projects/core/src/cms/store/selectors/components.selectors.ts [hunk 2]
```diff
@@ -36,37 +36,55 @@ export const componentsLoaderStateSelectorFactory = (
...
 export const componentsSelectorFactory = (
   uid: string,
   context: string
-): MemoizedSelector<StateWithCms, CmsComponent> => {
+): MemoizedSelector<StateWithCms, CmsComponent | null | undefined> => {
   return createSelector(
     componentsDataSelectorFactory(uid),
     componentsContextExistsSelectorFactory(uid, context),
     (componentState, exists) => {
+      console.log('uid', uid, exists);
```
Comment (Platonn) [inline projects/core/src/cms/store/selectors/components.selectors.ts]: 👀

File: projects/core/src/cms/store/selectors/components.selectors.ts [hunk 3]
```diff
@@ -36,37 +36,55 @@ export const componentsLoaderStateSelectorFactory = (
...
       if (componentState && exists) {
         return componentState;
       } else {
-        return undefined;
+        return exists === false ? null : undefined;
```
Comment (Platonn) [inline projects/core/src/cms/store/selectors/components.selectors.ts]: I have hard time trying to understand that. I'm hoping we can simplify above 4 lines to:
```
return exists ? componentState : null
```
What do you think about it?

I the light of that, I don't see the reason of change in the line 54.

But perhaps I'm missing something. Please help me understand.

File: projects/core/src/cms/store/selectors/components.selectors.ts (hunk 2 above)
Comment (dunqan) [reply projects/core/src/cms/store/selectors/components.selectors.ts]: thx, removed

File: projects/core/src/cms/store/selectors/components.selectors.ts (hunk 1 above)
Comment (dunqan) [reply projects/core/src/cms/store/selectors/components.selectors.ts]: 👍

File: projects/core/src/cms/store/selectors/components.selectors.ts (hunk 3 above)
Comment (dunqan) [reply projects/core/src/cms/store/selectors/components.selectors.ts]: we settled with
```
      switch (exists) {
        case true:
          return componentState;
        case false:
          return null;
        case undefined:
          return undefined;
      }
```

**Review by Platonn — APPROVED** [review]

**Pull request merged**